packages/salesforcedx-vscode-core/src/conflict/messages.ts

```diff
@@ -33,6 +35,36 @@ export const getConflictMessagesFor = (logName: string): ConflictDetectionMessag
             .build()
             .toString()
       }
+    ],
+    [
+      'retrieve_with_sourcepath',
+      {
+        warningMessageKey: retrieveWarningMessageKey,
+        commandHint: inputs =>
+          (Array.isArray(inputs) ? inputs : [inputs])
+            .map(input =>
+              new SfCommandBuilder()
+                .withArg('project:retrieve:start')
+                .withFlag('--sourcepath', input)
+                .build()
+                .toString()
+            )
+            .join('\n  ')
+      }
+    ],
+    [
+      'project_retrieve_start_default_scratch_org',
+      {
+        warningMessageKey: retrieveWarningMessageKey,
+        commandHint: () => nls.localize('pull_conflicts_error')
+      }
+    ],
+    [
+      'project_deploy_start_default_scratch_org',
+      {
+        warningMessageKey,
+        commandHint: () => nls.localize('push_conflicts_error')
+      }
     ]
   ]);
 
```

Comment:
this is a really strange fn.

Why not return the appropriate conflictMessage based on the log name?

instead of doing all the localization and commandBuilding, then building a map of all the possibilities, and then choosing to return only one of them?

And when could it ever return `undefined`?  It's going to throw otherwise?  But you could probably eliminate that at compile time by restricting `logName` to be one of a union type (`project_retrieve_start_default_scratch_org | ...`) so that there's no possibility of someone using it with an invalid key, so there's not this weird runtime error `No conflict messages found` that the user wouldn't be able to understand.